compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp

```diff
@@ -896,6 +898,35 @@ void QuantizeWithMinMaxPass::set_output_type(loco::Graph *g) const
   }
 }
 
+LayerInfoMap QuantizeWithMinMaxPass::create_layer_info_map(loco::Graph *g) const
+{
+  auto info_by_name = LayerInfoMap();
+
+  for (auto &&info : _ctx->layers_info)
+  {
+    auto name = info.name;
+    bool found = false;
+    for (auto node : loco::active_nodes(loco::output_nodes(g)))
+    {
+      auto cnode = loco::must_cast<luci::CircleNode *>(node);
+      if (cnode->name() == name)
+      {
+        info_by_name[name] = &info;
+        found = true;
+        break;
+      }
+    }
+
+    if (not found)
+      throw std::runtime_error("No such layer named " + name +
+                               ". Check layer names in the quantization configuration file.");
+  }
+
+  assert(info_by_name.size() == _ctx->layers_info.size()); // FIX_ME_UNLESS
+
+  return info_by_name;
```

Comment:
I'm not sure how big size memory would be for return but how about
```cxx
bool QuantizeWithMinMaxPass::create_layer_info_map(loco::Graph *g, LayerInfoMap &info_by_name) const
```
return false if not found or true if OK.

Reply:
AFAIK, std::map (and other std containers) will not be copied due to copy elision.